jerry-core/lit/lit-literal-storage.cpp

```diff
@@ -13,336 +14,86 @@
-  {
-    if (it_this.read<lit_utf8_byte_t> () != str[i])
-    {
-      return false;
-    }
-
-    it_this.skip<lit_utf8_byte_t> ();
-  }
-
-  return get_length () == str_size;
-} /* lit_charset_record_t::equal_non_zt */
-
-/**
- * Dump charset record to specified snapshot buffer
- *
- * @return number of bytes dumped,
- *         or 0 - upon dump failure
- */
-uint32_t
-lit_charset_record_t::dump_for_snapshot (uint8_t *buffer_p, /**< buffer to dump to */
-                                         size_t buffer_size, /**< buffer size */
-                                         size_t *in_out_buffer_offset_p) /**< in-out: buffer write offset */
-{
-  lit_utf8_size_t length = get_length ();
-  if (!jrt_write_to_buffer_by_offset (buffer_p, buffer_size, in_out_buffer_offset_p, length))
-  {
-    return 0;
-  }
-
-  rcs_record_iterator_t it_this (&lit_storage, this);
-
-  it_this.skip (header_size ());
-
-  for (lit_utf8_size_t i = 0; i < length; i++)
-  {
-    lit_utf8_byte_t next_byte = it_this.read<lit_utf8_byte_t> ();
-    it_this.skip<lit_utf8_byte_t> ();
-
-    if (!jrt_write_to_buffer_by_offset (buffer_p, buffer_size, in_out_buffer_offset_p, next_byte))
-    {
-      return 0;
-    }
-  }
-
-  return ((uint32_t) sizeof (length) + length * (uint32_t) sizeof (lit_utf8_byte_t));
-} /* lit_charset_record_t::dump_for_snapshot */
-
-/**
- * Dump number record to specified snapshot buffer
- *
- * @return number of bytes dumped,
- *         or 0 - upon dump failure
- */
-uint32_t
-lit_number_record_t::dump_for_snapshot (uint8_t *buffer_p, /**< buffer to dump to */
-                                        size_t buffer_size, /**< buffer size */
-                                        size_t *in_out_buffer_offset_p) const /**< in-out: buffer write offset */
-{
-  /* dumping as double (not ecma_number_t), because ecma_number_t can be float or double,
-   * depending on engine compile-time configuration */
-  double num = get_number ();
-  if (!jrt_write_to_buffer_by_offset (buffer_p, buffer_size, in_out_buffer_offset_p, num))
-  {
-    return 0;
-  }
+#include "ecma-helpers.h"
+#include "rcs-allocator.h"
+#include "rcs-iterator.h"
+#include "rcs-records.h"
 
-  return ((uint32_t) sizeof (num));
-} /* lit_number_record_t::dump_for_snapshot */
+record_set_t rcs_lit_storage;
 
 /**
  * Create charset record in the literal storage
  *
- * @return  pointer to the created record
+ * @return pointer to the created record
  */
-lit_charset_record_t *
-lit_literal_storage_t::create_charset_record (const lit_utf8_byte_t *str, /**< string to be placed in the record */
-                                              lit_utf8_size_t buf_size) /**< size in bytes of the buffer which holds the
-                                                                         * string */
+record_t *
+create_charset_record (record_set_t *rec_set_p, /**< recordset */
+                       const lit_utf8_byte_t *str, /**< string to be placed in the record */
```

Comment:
Two things:
1) str -> str_p
2) imho 'placed in the record' -> 'placed into the record'